packages/syft/src/syft/core/tensor/smpc/mpc_tensor.py

```diff
@@ -203,8 +203,17 @@ def publish(self, sigma: float) -> MPCTensor:
             share.block
 
         for share in self.child:
+            client = share.client
+            privacy_budget = client.privacy_budget  # before publish
             new_share = share.publish(sigma=sigma)
             new_shares.append(new_share)
+            new_share.block
+            curr_budget = client.privacy_budget  # budget after publish
+            if privacy_budget == curr_budget:
```

Comment:
This is actually a broken aspect of the current codebase - the privacy budget should always appear to change.

Reply:
The only way to estimate whether the .publish() asked for too much budget is shapiro() calculation.

Reply:
https://app.shortcut.com/openmined/story/4902/bug-privacy-budget-warning

Reply:
Modified to Shapiro by @fiza11